src/esp/gfx/SensorInfoVisualizer.cpp

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+#include "SensorInfoVisualizer.h"
+
+#include <Corrade/Utility/Assert.h>
+#include <Magnum/GL/DefaultFramebuffer.h>
+#include <Magnum/GL/RenderbufferFormat.h>
+#include <Magnum/Magnum.h>
+#include <Magnum/Math/Vector4.h>
+#include <Magnum/Shaders/Generic.h>
+
+#include "DepthVisualizerShader.h"
+
+namespace esp {
+namespace gfx {
+
+namespace Mn = Magnum;
+namespace Cr = Corrade;
+
+SensorInfoVisualizer::SensorInfoVisualizer() {
+  // prepare a big triangle mesh to cover the screen
+  mesh_ = Mn::GL::Mesh{};
+  mesh_.setCount(3);
+}
+void SensorInfoVisualizer::resetFramebufferRenderbuffer(
+    Mn::Vector2i framebufferSize) {
+  if (frameBuffer_.viewport() != Mn::Range2Di{{}, {framebufferSize}}) {
+    frameBuffer_ = Mn::GL::Framebuffer{{{}, framebufferSize}};
+    colorBuffer_.setStorage(Mn::GL::RenderbufferFormat::RGBA8, framebufferSize);
+    frameBuffer_.attachRenderbuffer(Mn::GL::Framebuffer::ColorAttachment{0},
+                                    colorBuffer_);
+    // depth buffer is 24-bit integer pixel
+    depthBuffer_.setStorage(Mn::GL::RenderbufferFormat::DepthComponent24,
+                            framebufferSize);
+    frameBuffer_.attachRenderbuffer(
+        Mn::GL::Framebuffer::BufferAttachment::Depth, depthBuffer_);
```

Comment:
Do you actually need the depth buffer? Isn't it only rendering a single textured triangle, no matter what kind of data gets visualized?

Reply:
I need a depth buffer later if I would like to "interact“. For example, if I would like to select and highlight some objects. ( I need to draw these objects on top of this framebuffer.)

Reply:
I need a depth buffer later if I would like to "interact“. For example, if I would like to select and highlight some objects. ( I need to draw these objects on top of this framebuffer.)